src/tests/ftest/util/network_utils.py

```diff
@@ -0,0 +1,560 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import errno
+from logging import getLogger
+import os
+import re
+
+from ClusterShell.NodeSet import NodeSet
+
+from exception_utils import CommandFailure
+from general_utils import run_task, display_task
+
+SUPPORTED_PROVIDERS = (
+    "ofi+sockets", "ofi+tcp", "ofi+tcp;ofi_rxm", "ofi+verbs", "ofi+verbs;ofi_rxm", "ucx+dc_x")
+
+
+class NetworkDevice():
+    """A class to represent the information of a network device."""
+
+    def __init__(self, host, device, ib_device, port, provider, numa):
+        """Initialize the network device data object."""
+        self.host = host
+        self.device = device
+        self.ib_device = ib_device
+        self.port = port
+        self.provider = provider
+        self.numa = numa
+
+    def __repr__(self):
+        """Overwrite to display formatted devices."""
+        return self.__str__()
+
+    def __str__(self):
+        """Overwrite to display formatted devices."""
+        settings = ["{}={}".format(key, getattr(self, key, None)) for key in self.__dict__]
+        return "NetworkDevice({})".format(", ".join(settings))
+
+    def __ne__(self, other):
+        """Override the default not-equal implementation."""
+        return not self.__eq__(other)
+
+    def __eq__(self, other):
+        """Override the default implementation to compare devices."""
+        status = True
+        if not isinstance(other, NetworkDevice):
+            return False
+        for key in self.__dict__:
+            try:
+                status &= str(getattr(self, key)) == str(getattr(other, key))
+            except AttributeError:
+                return False
+        return status
+
+    @property
+    def domain(self):
+        """Get the domain for this network device.
+
+        Returns:
+            str: the domain for this network device
+
+        """
+        if "ucx" in self.provider:
+            return ":".join([self.ib_device, self.port])
+        return self.ib_device
+
+
+def get_active_network_interfaces(hosts, verbose=True):
+    """Get all active network interfaces on the hosts.
+
+    Args:
+        hosts (NodeSet): hosts on which to find active interfaces
+        verbose (bool, optional): display command details. Defaults to True.
+
+    Returns:
+        dict: a dictionary of interface keys and NodeSet values on which they were found
+
+    """
+    net_path = os.path.join(os.path.sep, "sys", "class", "net")
+    operstate = os.path.join(net_path, "*", "operstate")
+    command = " | ".join(
+        ["grep -l 'up' {}".format(operstate), "grep -Ev '/(lo|bonding_masters)/'", "sort"])
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string